**L'objectif ici est de comparer les notes données à la culture et la présence ou non de lieux culturels par arrondissement.**

In [104]:
!pip install openpyxl
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Commençons par nous intéresser aux notes pour la culture de chaque arrondissement :

In [105]:
# Création d'un dataframe NotesCulture contenant les notations pour la culture de chaque arrondissement
Notes = pd.read_excel('/home/onyxia/Projet_Python_2AENSAE/Notes_arrondissements.xlsx', decimal=',', index_col=0)
NotesCulture = Notes['Culture']
NotesCulture

Paris 1er    8.58
Paris 2e     8.00
Paris 3e     9.33
Paris 4e     7.46
Paris 5e     8.50
Paris 6e     8.40
Paris 7e     7.24
Paris 8e     7.44
Paris 9e     8.50
Paris 10e    6.18
Paris 11e    7.01
Paris 12e    6.90
Paris 13e    6.76
Paris 14e    6.54
Paris 15e    6.72
Paris 16e    6.66
Paris 17e    6.65
Paris 18e    5.89
Paris 19e    7.01
Paris 20e    6.59
Name: Culture, dtype: float64

In [106]:
# Affichage des 4 meilleurs et des 4 pires arrondissement pour leur culture, d'après les notes des habitants
print('Les 4 arrondissements ayant les meilleures notes pour leur culture sont :')
display(NotesCulture.sort_values(ascending=False)[:4])

print('Les 4 arrondissements ayant les pires notes pour leur culture sont :')
display(NotesCulture.sort_values()[:4])

Les 4 arrondissements ayant les meilleures notes pour leur culture sont :


Paris 3e     9.33
Paris 1er    8.58
Paris 9e     8.50
Paris 5e     8.50
Name: Culture, dtype: float64

Les 4 arrondissements ayant les pires notes pour leur culture sont :


Paris 18e    5.89
Paris 10e    6.18
Paris 14e    6.54
Paris 20e    6.59
Name: Culture, dtype: float64

En ce qui concerne les lieux culturels, les Parisiens semblent donc avoir une préférence pour les arrondissements 3, 1, 9 et 5.
Le troisième arrondissement est largement en tête avec une note de 9.33.
Le premier, neuvième et cinquième arrondissement forment un deuxieme groupe en tête avec une note autour de 8.5.

Les arrondissements 18, 10, 14 et 20 obtiennent eux les pires notes : leurs notes sont réparties entre 5.89 pour le 18ème et 6.59.

Voyons quelques statistiques sur les notes données pour la culture :

In [107]:
NotesCulture.describe()

count    20.000000
mean      7.318000
std       0.931725
min       5.890000
25%       6.657500
50%       7.010000
75%       8.100000
max       9.330000
Name: Culture, dtype: float64

Ces premières statistiques descriptives nous apprennent que le top 4 des arrondissement se situent au dessus de 75% des notes, ce qui est normal mathématiquement.
La note moyenne obtenue pour la culture est de 7.32, ce qui est haut. L'écart-type est de 0.93 ce qui est assez bas. Les notes sont
plutôt concentrées autour de 7. 

Si on compare donc les pires et meilleures notes, c'est surtout le 3ème arrondissement qui se démarque avec son 9.33. La pire note, 5.89, n'est pas autant éloignée du groupe que cela.

Maintenant, nous allons importer la base de données recensant tous les lieux culturels parisiens. Nous pourrons alors voir s'il y a une corrélation entre le nombre de lieux culturels et la satisfaction des habitants parisiens.

### Import de la base de données recensant les divers lieux culturels de la capitale

In [108]:
# Import de la base de données téléchargée sur data.gouv.fr
LieuxCulturels = pd.read_csv('/home/onyxia/Projet_Python_2AENSAE/lieux_culturels.csv')
display(LieuxCulturels)

NOM_LIEU   DEP  \
0         Acad‚mie de Danse Barbot Perrot - Danse  75.0   
1                             Acad‚mie de Guitare  75.0   
2                     Association Cours Hippolyte  75.0   
3                              Atelier d'Artistes  75.0   
4     Atelier de Recherche Chor‚graphique - Danse  75.0   
...                                           ...   ...   
1102                                Ya d'la Voix!  75.0   
1103                          Yamaha Music School  75.0   
1104                                 Yvon Lambert  75.0   
1105               Archives Nationales Site Paris  75.0   
1106             Archives du d‚partement de Paris  75.0   

                       COMMUNE  CODE_INSEE  CODE_POSTAL  RAYONNEMENT GPE_800M  \
0     Paris 17e Arrondissement     75117.0      75017.0          3.0      Oui   
1     Paris 13e Arrondissement     75113.0      75013.0          3.0      Oui   
2     Paris 17e Arrondissement     75117.0      75017.0          3.0      Oui   
3     Paris 17e Arrondissement     75117.0      75017.0          3.0      Oui   
4     Paris 13e Arrondissement     75113.0      75013.0          3.0      Oui   
...                        ...         ...          ...          ...      ...   
1102  Paris 11e Arrondissement     75111.0      75011.0          3.0      Non   
1103  Paris 12e Arrondissement     75112.0      75012.0          3.0      Non   
1104   Paris 3e Arrondissement     75103.0      75003.0          3.0      Non   
1105   Paris 3e Arrondissement     75103.0      75003.0          1.0      Non   
1106  Paris 19e Arrondissement     75119.0      75019.0          1.0      Non   

                       GPE_STATION  GPE_LIGNE CARTO_BIB  ...  C_SUF3 C_DESI  \
0                    Pont Cardinet  GPE - L14       NaN  ...     NaN    RUE   
1     Maison Blanche - Paris XIIIe  GPE - L14       NaN  ...     NaN    PAS   
2                    Pont Cardinet  GPE - L14       NaN  ...     NaN    RUE   
3                    Pont Cardinet  GPE - L14       NaN  ...     NaN    RUE   
4     Maison Blanche - Paris XIIIe  GPE - L14       NaN  ...     NaN     AV   
...                            ...        ...       ...  ...     ...    ...   
1102                           NaN        NaN       NaN  ...     NaN    RUE   
1103                           NaN        NaN       NaN  ...     NaN    RUE   
1104                           NaN        NaN       NaN  ...     NaN    RUE   
1105                           NaN        NaN       NaN  ...     NaN    RUE   
1106                           NaN        NaN       NaN  ...     NaN     BD   

      C_LIAISON                  L_VOIE      N_SQ_EP C_NIV1 C_NIV2   C_NIV3  \
0            DE             TOCQUEVILLE  753000955.0   13.0  132.0  13211.0   
1            DU                MOULINET  753000741.0   13.0  132.0  13211.0   
2           NaN                CARDINET  753000238.0   13.0  132.0  13211.0   
3           DES                   DAMES  753000100.0   13.0  134.0  13413.0   
4            DE                  CHOISY  753000625.0   13.0  132.0  13211.0   
...         ...                     ...          ...    ...    ...      ...   
1102         DU  FAUBOURG SAINT-ANTOINE  753000231.0   13.0  132.0  13211.0   
1103         DE                   BERCY  753000573.0   13.0  132.0  13211.0   
1104        NaN       VIEILLE DU TEMPLE  753000707.0   13.0  133.0  13321.0   
1105        DES        FRANCS BOURGEOIS  751009138.0   13.0  131.0  13121.0   
1106        NaN                SERURIER  751005470.0   13.0  131.0  13121.0   

           COORD_X       COORD_Y  
0     6.496656e+09  6.864963e+09  
1     6.525824e+09  6.858678e+09  
2     6.494944e+09  6.865273e+09  
3     6.498860e+09  6.864952e+09  
4     6.528911e+09  6.858847e+09  
...            ...           ...  
1102  6.545428e+09  6.861345e+09  
1103  6.545667e+09  6.860013e+09  
1104  6.532523e+09  6.862514e+09  
1105  6.528442e+09  6.862406e+08  
1106  6.564877e+09  6.864493e+09  

[1107 rows x 50 columns]

### Préparation de cette base de données pour notre étude

In [109]:
# Suppression des '.0' dans la liste des codes postaux par soucis d'esthétisme :
LieuxCulturels.CODE_POSTAL = LieuxCulturels.CODE_POSTAL.astype(str)
LieuxCulturels.CODE_POSTAL = LieuxCulturels.CODE_POSTAL.str.replace('.0','')

In [110]:
# Suppression des lignes n'étant pas associées à un arrondissement :
LieuxCulturels.drop(LieuxCulturels[LieuxCulturels.CODE_POSTAL == 'nan'].index, inplace=True)

In [111]:
# Transformation de la base pour qu'elle recense uniquement le nombre de lieux culturels par arrondissement.
NbLieux = LieuxCulturels.CODE_POSTAL.value_counts()

In [112]:
# Affichage des 4 arrondissement ayant le plus (et le moins) de lieux culturels
print('Les 4 arrondissements ayant le plus de lieux culturels sont :')
display(NbLieux.sort_values(ascending=False)[:4])

print('Les 4 arrondissements ayant le moins de lieux culturels sont :')
display(NbLieux.sort_values()[:4])

Les 4 arrondissements ayant le plus de lieux culturels sont :


CODE_POSTAL
75011    90
75019    72
75018    71
75006    69
Name: count, dtype: int64

Les 4 arrondissements ayant le moins de lieux culturels sont :


CODE_POSTAL
75002    21
75007    29
75001    31
75003    47
Name: count, dtype: int64

Aucun des 4 arrondissements ayant les meilleures note pour leur culture ne se trouvent dans la liste des 4 arrondissements ayant le plus de lieux culturels. Il semble donc venir de cette analyse que la qualité prime sur la quantité de lieux culturels pour la satisfaction des habitants.

Les 4 arrondissements ayant les pires notes ne sont absolument pas ceux ayant le moins d'institutions culturelles non plus. La conclusion semble alors se confirmer. Cela va même encore plus loin : le 3ème arrondissement, qui est de loin le préféré des Parisiens sur le critère de la culture, est le 4ème arrondissement ayant le moins de lieux culturels !


##### Piste d'amélioration de l'analyse :

Plutot que de s'intéresser au nombre brut de lieux culturels par arrondissement; nous allons prendre la densité de lieux culturels comme variable d'intérêt. Nous allons donc inclure la superficie des arrondissements dans l'analyse pour voir si les résultats
obtenus sont davantage cohérents.

In [113]:
# Création d'une base de donnée associant pour chaque arrondissement sa superficie.

# Import des données géographiques des arrondissements parisiens depuis la librairie cartiflette

import cartiflette.s3 as s3

Arrondissements = s3.download_vectorfile_url_all(
    crs = 4326,
    values = ["75"],
    borders="COMMUNE_ARRONDISSEMENT",
    vectorfile_format="topojson",
    filter_by="DEPARTEMENT",
    source="EXPRESS-COG-CARTO-TERRITOIRE",
    year=2022)

# Stockage des superficies dans la series superficies

Arrondissements = Arrondissements.sort_values('INSEE_COG').reset_index()
superficies = Arrondissements.area
superficies

https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=75/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 40.1kiB [00:00, 18.2MiB/s]
/tmp/ipykernel_10202/3881435485.py:19: UserWarning: Geometry is in a geographic CRS. Results from 'area' are likely incorrect. Use 'GeoSeries.to_crs()' to re-project geometries to a projected CRS before this operation.

  superficies = Arrondissements.area


0     0.000224
1     0.000122
2     0.000143
3     0.000196
4     0.000312
5     0.000264
6     0.000501
7     0.000476
8     0.000267
9     0.000354
10    0.000447
11    0.002007
12    0.000874
13    0.000688
14    0.001038
15    0.002011
16    0.000695
17    0.000735
18    0.000832
19    0.000736
dtype: float64

Notes : nous ne nous interessons pas à l'unité de mesure des superficies, le rapport entre les unes et les autres nous suffira pour comparer les densités.

In [114]:
NbLieux.to_frame().sort_values('CODE_POSTAL').reset_index()

CODE_POSTAL  count
0        75001     31
1        75002     21
2        75003     47
3        75004     49
4        75005     51
5        75006     69
6        75007     29
7        75008     60
8        75009     63
9        75010     48
10       75011     90
11       75012     63
12       75013     60
13       75014     56
14       75015     57
15       75016     50
16       75017     60
17       75018     71
18       75019     72
19       75020     58

In [115]:
# Ajout des données de superficie au dataframe contenant le nombre de lieux culturels par arrondissement.

NbLieux = NbLieux.to_frame().sort_values('CODE_POSTAL').reset_index()
NbLieux['superficie'] = superficies.values
NbLieux = NbLieux.rename(columns={'count':'nb_lieux'})

In [120]:
# Création d'une colonne contenant les densités des lieux culturels 
NbLieux.nb_lieux = NbLieux.nb_lieux.astype('float64')
NbLieux['densite_culture'] = NbLieux.nb_lieux / NbLieux.superficie

In [121]:
display(NbLieux.sort_values('densite_culture', ascending=False))

CODE_POSTAL  nb_lieux  superficie  densite_culture
2        75003      47.0    0.000143    327629.860617
5        75006      69.0    0.000264    261588.562767
3        75004      49.0    0.000196    250480.366845
8        75009      63.0    0.000267    235769.799318
10       75011      90.0    0.000447    201122.088638
1        75002      21.0    0.000122    172625.823942
4        75005      51.0    0.000312    163652.468728
0        75001      31.0    0.000224    138314.302273
9        75010      48.0    0.000354    135459.567342
7        75008      60.0    0.000476    126101.970295
17       75018      71.0    0.000735     96553.249966
18       75019      72.0    0.000832     86531.762712
16       75017      60.0    0.000695     86359.480489
13       75014      56.0    0.000688     81438.510930
19       75020      58.0    0.000736     78854.053963
12       75013      60.0    0.000874     68613.745504
6        75007      29.0    0.000501     57882.357436
14       75015      57.0    0.001038     54908.932182
11       75012      63.0    0.002007     31390.783464
15       75016      50.0    0.002011     24857.943169

Le résultat est convaincant : le 3ème arrondissement, qui dispose de la meilleure note - et de loin - pour sa culture, est passé à la première place sur la classement des arrondissements disposant du plus de lieux culturels au mètre carré. Rappelons qu'il n'était même pas dans le top 4 des arrondissements disposant du plus de lieux culturels en nombre brut. 

Alors qu'elle n'apparaissait pas avant de considérer cette approche, nous pouvons maintenant supposer que la richesse du nombre de lieux culturels dans le troisième arrondissement explique en partie la note moyenne donnée par les habitants.

Les arrondissements 12 et 16 se retrouvent en bas de liste en terme de densité de lieux culturels : cela s'explique par le fait que la superficie prend en compte les bois de Boulogne et de Vincennes qui gonflent de façon énorme la valeur de leur superficie. 